kin-core/src/main/java/kin/core/KinAccount.java

```diff
@@ -104,6 +106,63 @@ TransactionId sendTransactionSync(@NonNull String publicAddress, @NonNull BigDec
     @NonNull
     Balance getBalanceSync() throws OperationFailedException;
 
+    /**
+     * Create {@link Request} for getting all the transactions payments history for the current account.
+     * <p> See {@link KinAccount#getPaymentsHistorySync()} for possibles errors</p>
+     *
+     * @return {@code Request<List<PaymentInfo>> } PaymentInfo - the payment information for a specific payment operation in a transaction
+     */
+    @NonNull
+    Request<List<PaymentInfo>> getPaymentsHistory();
+
+    /**
+     * For method description see {@link KinAccount#getPaymentsHistory()}
+     * @param requestParams an optional parameters to the request, such as accountId, cursor, limit, and order.
+     * Create {@link Request} for getting all the transactions payments history for a given account.
```

Comment:
I don't think we should expose the possibility of giving an account id, as it doesn't match the current interfaces - `KinAccount` is for a specific address and it doesn't appropriate to give an option here to query other accounts.